src/test/java/io/confluent/connect/elasticsearch/integration/ElasticsearchConnectorIT.java

```diff
@@ -138,4 +147,33 @@ public void testUpsert() throws Exception {
       }
     }
   }
+  
+  /*
+   * Currently writing primitives to ES fails because ES expects a JSON document and the connector
+   * does not wrap primitives in any way into a JSON document.
+   */
+  @Test
+  public void testBadRecordsInDLQ() throws Exception {
+    props.put(VALUE_CONVERTER_CLASS_CONFIG, StringConverter.class.getName());
+    props.put(BEHAVIOR_ON_MALFORMED_DOCS_CONFIG,"ignore");
+    props.put(DROP_INVALID_MESSAGE_CONFIG,"true");
+    props.put("errors.tolerance","all");
+    props.put("errors.deadletterqueue.topic.name","dlq_file_sink_02");
+    props.put("errors.deadletterqueue.topic.replication.factor", "1");
+    SinkTaskContext context = mock(SinkTaskContext.class);
+    ErrantRecordReporter reporter = mock(ErrantRecordReporter.class);
+    when(context.errantRecordReporter()).thenReturn(reporter);
+    helperClient = new ElasticsearchHelperClient(new ElasticsearchClient(new ElasticsearchSinkConnectorConfig(props), reporter).client());
+
+    connect.configureConnector(CONNECTOR_NAME, props);
+    
+    // wait for tasks to spin up
+    waitForConnectorToStart(CONNECTOR_NAME, TASKS_MAX);
+    
+    for (int i  = 0; i < NUM_RECORDS; i++) {
+      connect.kafka().produce(TOPIC, String.valueOf(i),  null);
+    }
+    
+    waitForRecords(0);
```

Comment:
you should also consume from the dlq topic and make sure it actually contains the records you expect